build-system/tasks/e2e/describes-e2e.js

```diff
@@ -204,23 +204,73 @@ class AmpPageFixture {
     capabilities.set('chromeOptions', chromeOptions);
 
     const builder = new Builder().withCapabilities(capabilities);
-    return builder.build().then(driver => {
-      const controller = new SeleniumWebDriverController(driver);
-      env.controller = controller;
-      env.ampDriver = new AmpDriver(controller);
-      this.driver_ = driver;
-    });
+    const driver = await builder.build();
+    const controller = new SeleniumWebDriverController(driver);
+    const ampDriver = new AmpDriver(controller);
+
+    env.controller = controller;
+    env.ampDriver = ampDriver;
+    this.driver_ = driver;
+
+    const {
+      fixture,
+      experiments,
+      environment,
+      serveMode,
+    } = this.spec;
+
+    await setServeMode(ampDriver, serveMode);
+    await toggleExperiments(ampDriver, fixture, experiments);
+    await ampDriver.navigateToEnvironment(environment, fixture);
```

Comment:
What's the difference between `environment` and `fixture`?

Reply:
`environment` is the outer document that the ampdoc is loaded into, and its values are `single`, `viewer-demo` and `shadow-demo`. For now I added the demo suffix since I reused the existing example pages and I think we would want separate pages for general example testing and e2e testing.

`fixture` is the URL of the amp document to load. I'm open to suggestions for any of these property names if you have them!

Reply:
Got it. So in my mind `environment` and `fixture` mean roughly the same thing - the context in which tests are run - that `describes-e2e.js` sets up. There's an existing example, `AmpPageFixture`, that `describes-e2e.js` sets up the `single` AMP environment. I think `viewer-demo` and `shadow-demo` should also be fixtures, to follow that pattern.

If that sounds okay, we can rename the `fixture` parameter to `testUrl` or something like that.

Reply:
That makes sense to me, let's go with that!